test/exec/test_at_coroutine_exit.cpp

```diff
@@ -0,0 +1,239 @@
+  REQUIRE(result == 5);
+}
+
+namespace {
+struct test_exception : std::runtime_error {
+  using std::runtime_error::runtime_error;
+};
+
+exec::task<void> invoke_action_after_exception(int& result) {
+  ++result;
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  throw test_exception("test");
+  ++result;
+}
+}
+
+TEST_CASE("at_coroutine_exit invokes two actions in correct order after exception", "[task][at_coroutine_exit]")
+{
+  int result = 0;
+  REQUIRE_THROWS_AS(stdexec::sync_wait(invoke_action_after_exception(result)), test_exception);
+  REQUIRE(result == 3);
+}
+
+namespace {
+exec::task<void> invoke_actions_after_stop(int& result, auto stop) {
+  ++result;
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  co_await stop;
+  ++result;
+}
+
+exec::task<void> invoke_actions_after_stop2(int& result, auto stop) {
+  ++result;
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+
+  co_await stop;
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  ++result;
+}
+
+exec::task<void> invoke_actions_after_stop3(int& result, auto stop) {
+  ++result;
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  ++result;
+
+  co_await stop;
+}
+
+exec::task<void> invoke_actions_after_stop4(int& result, auto stop) {
+  ++result;
+  co_await stop;
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result += 1;
+    co_return;
+  }, result);
+
+  co_await exec::at_coroutine_exit([](int& result) -> exec::task<void> {
+    result *= 2;
+    co_return;
+  }, result);
+  ++result;
+}
+}
+
+TEST_CASE("at_coroutine_exit invokes two actions in correct order after stop signal", "[task][at_coroutine_exit]")
+{
+  int result = 0;
+  stopped_scheduler scheduler;
+  stdexec::sync_wait(invoke_actions_after_stop(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 3);
+
+  result = 0;
+  stdexec::sync_wait(invoke_actions_after_stop2(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 2);
+
+  result = 0;
+  stdexec::sync_wait(invoke_actions_after_stop3(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 5);
+
+  result = 0;
+  stdexec::sync_wait(invoke_actions_after_stop4(result, stdexec::schedule(scheduler)));
+  REQUIRE(result == 1);
+}
+
+
+#ifdef __linux__
```

Comment:
I'm curious why this is restricted to `__linux__`.

Reply:
`__linux__` is not accurate. I think one would need to check in CMake if calling those POSIX functions does compile and define some Macro at configure time. As a workaround I could just check the existence of `<unistd.h>` and `<sys/wait.h>`  with  `__has_include`